test/worker_versioning_test.go

```diff
@@ -334,7 +334,7 @@ func (ts *WorkerVersioningTestSuite) TestCommitRules() {
 	ts.Equal(1, len(res.AssignmentRules))
 	ts.Equal("2.0", res.AssignmentRules[0].Rule.TargetBuildID)
 	_, ok := res.AssignmentRules[0].Rule.Ramp.(*client.VersioningRampByPercentage)
-	ts.Falsef(ok, "Still has a percentage ramp")
+	ts.Truef(ok, "Ramp at 100 but type isn't updated")
```

Comment:
I am not sure exactly what changed in server to cause this change in behavior, so this could be an insufficient bandaid fix. 

Printing the value of *Ramp, it prints `100`. If I understand the Go flow here, `validateRamp()` should return an Error from this line? `return errors.New("invalid percentage in `Ramp`, not in [0,100)")`, which should cause `res.AssignmentRules[0].Rule.Ramp` to not be type `*client.VersioningRampByPercentage` anymore?